roles/jd-client/src/lib/job_declarator/mod.rs

```diff
@@ -55,7 +55,7 @@ pub struct JobDeclarator {
     req_ids: Id,
     min_extranonce_size: u16,
     // (Sent DeclareMiningJob, is future, template id, merkle path)
-    last_declare_mining_jobs_sent: HashMap<u32, Option<LastDeclareJob>>,
+    last_declare_mining_jobs_sent: [Option<(u32, LastDeclareJob)>; 2],
```

Comment:
Can you explain why is ok to have at max 2 last_declare_mining_job and at min 0. I would like to see in which cases we have 0, when 1 when 2 and most important why more then 2 is impossible

Reply:
f the system detects more than 2 jobs, it should shut down to avoid potential problems. This helps prevent issues like downstream components sending shares for outdated jobs, which can lead to inefficiencies and financial losses. I'm not entirely certain about this, but we don't want miners wasting energy if something unusual occurs. I am still looking for concrete proof. This assertion is based on your response here: [GitHub Discussion](https://github.com/stratum-mining/stratum/pull/904#discussion_r1609469048).

Reply:
Can you add a comment in the code for it? It would be nice to state also way we can't have more then 2 jobs. (I don't remember why, if you can't figure out I can take a look)

Reply:
Maybe @GitGab19 have more context?

Reply:
The context is the one already referenced. If JDC is using more than 2 jobs it means something wrong is happening. The edge case I'm thinking about is the one in which more than 2 blocks are found in some seconds. So new templates and new declared mining jobs are created by JDC. How this edge case is handled in this PR?

Reply:
Perfect, so I think that you just need to put this as a comment in the code and this PR is ready to go

Reply:
@GitGab19, I have added the comment.